shared/src/pool_aggregating.rs

```diff
@@ -11,53 +11,35 @@ use std::sync::Arc;
 use structopt::clap::arg_enum;
 
 arg_enum! {
-    #[derive(Debug, Clone)]
+    #[derive(Debug, Clone, Copy, Eq, Hash, PartialEq)]
     pub enum BaselineSources {
         Uniswap,
         Sushiswap,
     }
 }
 
-pub async fn pair_providers(
-    sources: &[BaselineSources],
+pub async fn pair_provider(
+    source: BaselineSources,
```

Comment:
nit: Maybe this should be singular now
```suggestion
    source: BaselineSource,
```